src/algorithms/MPolyEvaluate.jl

```diff
@@ -0,0 +1,289 @@
+###############################################################################
+#
+#   MPolyEvaluate.jl : Various algorithms for evaluating multivariates
+#
+###############################################################################
+
+function add!(z::T, a, b)::T where T
+  return a + b
+end
+
+function mul!(z::T, a, b)::T where T
+  return a*b
+end
+
+function pow!(z::T, a, b)::T where T
+  return a^b
+end
+
+# a *= x^p, using t as temp. Must return the modifed a and t
```

Comment:
Must `t == x^p` hold?

Reply:
of course not

Reply:
Why must it be returned then? I don't quite understand. Can you elaborate?